### PR TITLE
fix: Check IntersectionObserver 

### PR DESCRIPTION
**Pull request by chefilip**
Screen Shot 2022-03-01 at 15 12 41

### CODE REVIEW

**Comment by memoyil** [comment]
Does this happen on serverside rendering?

**Comment by chefilip** [comment]
> Does this happen on serverside rendering?

The error is currently triggered on client side. But it would be a good practice to put `IntersectionObserver` in `useLayoutEffect` instead of `useEffect`

File: src/hooks/useIntersectionObserver.ts [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
   const observerRef = useRef((element) => setObserverRefElement(element))
   const intersectionObserverRef = useRef<IntersectionObserver>(null)
   const [isIntersecting, setIsIntersecting] = useState(false)
 
-  useEffect(() => {
-    if (!intersectionObserverRef.current && observerRefElement) {
+  useLayoutEffect(() => {
+    const isSupported = typeof window === 'object' && window.IntersectionObserver
+
+    if (isSupported && !intersectionObserverRef.current && observerRefElement) {
```
Comment (chef-jojo) [inline src/hooks/useIntersectionObserver.ts:12]: if not isSupported we should just say it intersecting?

Comment (chefilip) [reply src/hooks/useIntersectionObserver.ts:12]: Hmm. I think it would be fine to do that. Will add it

**Review by chef-jojo — APPROVED** [review]

**Pull request merged**